### ISSUE TITLE
Number keys show their shifted symbols (e.g. `1` and `!`), while symbol keys do not (e.g. `=`, `[`)

### ISSUE THREAD

**Issue by rca** [keyboardio/Chrysalis#611]
**Is your feature request related to a problem? Please describe.**

When editing a layout in Chrysalis, the number keys show the symbol that they would print when the shift key is held down, while symbol keys do not.

**Describe the solution you'd like**

It would be nice to see the shifted key values for symbols just like they are visible for numbers.

**Comment by algernon** [comment]
I believe this is now fixed on master, with the new editor & keycode database. Can you verify if the current snapshot  does what you think it should do?

**Comment by aspiers** [comment]
It looks to me with 0.8.1 that the opposite is now true: *neither* the number keys nor symbol keys show what they will output when shift is held.

**Comment by algernon** [comment]
Oooh, hmm. You mean that the _selector_ should show _both_? (As opposed to the keymap showing the shifted symbol if the key is mapped to shift+key)

**Comment by aspiers** [comment]
I'm not sure what you mean by selector, but as per the original report this issue is about what is displayed on the graphical representation of the keycaps.

**Comment by algernon** [comment]
By selector, I mean the 104-key keyboard that pops up when you press "Pick a key".

Nevertheless, the other part of my question is still relevant: should the keycaps display both `1` and `!` for a key mapped to `1`, (but only `!` when mapped to `Shift+1`)? Right now, they display `1` when mapped to `1`, and `!` when mapped to `Shift+1`. The reason behind this is that right now, we display what keys do on their own, without modifiers or anything fancy. If we added modifiers, we'd run out of space quickly, because we'd need Shift and AltGr displayed for some keys too.

**Comment by aspiers** [comment]
[**@algernon**](https://github.com/algernon) commented on [January 26, 2021 6:09 PM](https://github.com/keyboardio/Chrysalis/issues/611#issuecomment-767727480 "2021-01-26T18:09:42Z - Replied by Github Reply Comments"):
> By selector, I mean the 104-key keyboard that pops up when you press "Pick a key".

OK, so we're talking about the same thing.

> Nevertheless, the other part of my question is still relevant: should the keycaps display both `1` and `!` for a key mapped to `1`, (but only `!` when mapped to `Shift+1`)?

Yes.  As witnessed by the title of this issue, this is how it was before your jumbo PR (which improved a ton of things BTW - congrats on that!)  And this makes sense because a) that's what you typically see on the keycaps so everyone understands what it means, and b) there's enough space to show it, and it makes life easier, so why not?

> Right now, they display `1` when mapped to `1`, and `!` when mapped to `Shift+1`. The reason behind this is that right now, we display what keys do on their own, without modifiers or anything fancy. If we added modifiers, we'd run out of space quickly, because we'd need Shift and AltGr displayed for some keys too.

Please let's not throw the baby out with the bathwater ;-)  Displaying the shifted symbols is better than not displaying them.  If there isn't room to display other stuff too, that doesn't invalidate the benefits of displaying the shifted symbols.

**Comment by algernon** [comment]
Showing them on the 104-key selector is certainly something we can do, I have nothing against that. Can likely fit AltGr there too, for good measures.

It's the keyboard image where I'd be _much_ more conservative about what we display and what we don't.

**Comment by aspiers** [comment]
OK so maybe I misunderstood.  I don't know what the difference is between the 104-key selector and the keyboard image.  All I know is that, as per this issue title, prior to the big redesign, the number keys showed the shifted symbols but the symbol keys didn't, and now neither do.  IMHO (and presumably in the eyes of the OP) this is a move in the opposite direction from what would be ideal.

**Comment by Rom1deTroyes** [comment]
I have the same feelings here (from both of you :thinking: :
- if the 104 key picker is meant to be an image of a real keyboard, I expect to see the same symbols as my laptop keyboard (ie. <kbd>AZERTY</kbd>... for the letters, <kbd>&1</kbd><kbd>é2~</kbd><kbd>"3#</kbd>... for the numbers, <kbd>)°]</kbd><kbd>=+}</kbd><kbd>^¨</kbd><kbd>$£ê</kbd>... for the symbols, etc.) : the cargo picker is an awesome improvement here,and choosing the system layout is a life changer !
- On the main keyboard image (the one with the Model01 or the Atreus), I expect to see just what will happen when I press the key, so per layer representation as it is now seems very intuitive for me, but needed to be tweaked at the corners.

For now, if I want to have a layer with the legacy numbers row splited like numbers on the first row and symbols on the second row, and my system is in AZERTY, here is the screen I have :

Chrysalis-0.7.7 :
- the letters on the Atreus keyboard are from the QWERTY keyboard,
- the numbers (shifted in AZERTY) show both lower and upper character.
![Chrysalis-0.7.7](https://user-images.githubusercontent.com/8625432/105908569-a77ddf00-6026-11eb-8afc-9ff2a31bec92.png)

Chrysalis-0.8.2-snapshot :
- the numbers (shifted in AZERTY) on the Atreus keyboard are the shifted symbols from QUERTY, and on the 104 picker the upper symbols are not shown (so you can't find `+%?./`.
- there is also problem with <kbd>éèçà</kbd> that are shown in an upper font, but are not shifted (`É` is <kbd>AltGr</kbd>+<kbd>Shift</kbd>+<kbd>é</kbd> on my GNU/Linux box).
![Chrysalis-0.8.2-snapshot](https://user-images.githubusercontent.com/8625432/105908790-ee6bd480-6026-11eb-89b5-0f64748948d1.png)

I don't know if you prefer having all the ideas and feedbacks here to think about it, or splitting things on others issues to clarify all this assaults :-)

Amazing work, BTW !

**Comment by obra** [comment]
I agree that we should show (at least) shifted symbols (but not both upper and lower case letters) on the key picker.

**Comment by Rom1deTroyes** [comment]
Yes, showing both <kbd>aA</kbd>, <kbd>bB</kbd> <kbd>cC</kbd> is very heavy and don't add simplicity !

Some examples, for reference ; 

- A heavy but complete technical picture : ![image](https://user-images.githubusercontent.com/8625432/105917144-45c37200-6032-11eb-9b01-a4bc1a76024f.png)

- a complete keyboard, without lower case when no need to : ![image](https://user-images.githubusercontent.com/8625432/105916589-71922800-6031-11eb-903e-250ce873ecbf.png)

- a good compromise between light and complete : ![image](https://user-images.githubusercontent.com/8625432/105916481-414a8980-6031-11eb-9eac-7a717c6097ad.png)

- Some users may need a more than complete option : ![image](https://user-images.githubusercontent.com/8625432/105916691-a0100300-6031-11eb-9f8a-6c9aaea58c87.png)

- An other way to look at layers on a standard keyboard : ![image](https://user-images.githubusercontent.com/8625432/105917082-347a6580-6032-11eb-87bc-c13b260f7c91.png)

**Comment by aspiers** [comment]
Ah OK, now I finally understand what you mean by the selector / picker.  I hadn't noticed that until these comments.  But this issue was originally about the main keyboard, before this picker even existed.  So while it's great to discuss both, please let's not lose sight of the original issue and the fact that the recent redesign has moved in the opposite direction from what the original report was asking for.

**Comment by eloyesp** [comment]
@Rom1deTroyes Can you share how you generate those images?

I really think that Chrysalys needs to make it possible to generate those kind of images, not by adding the functionality, but integrating with some other software. So Chrysalis can focus on one task, that is customizing the layout, and then another software can focus on customize the print version, share images, etc. I think that this relates somehow with #268.
